avocado/core/runners/sysinfo.py

```diff
@@ -0,0 +1,311 @@
+import logging
+import multiprocessing
+import os
+import time
+import traceback
+from abc import ABC, abstractmethod
+
+from avocado.core import nrunner
+from avocado.core.runners.utils import messages
+from avocado.core.settings import settings
+from avocado.utils import astring, process, software_manager
+
+log = logging.getLogger("avocado.sysinfo")
+
+
+class _Collectible(ABC):
+
+    """
+    Abstract class for representing collectibles by sysinfo.
+    """
+
+    def __init__(self, logf):
+        self.logf = astring.string_to_safe_path(logf)
+
+    @abstractmethod
+    def run(self, queue, logdir):
+        pass
+
+
+class _Logfile(_Collectible):
+
+    """
+    Collectible system file.
+
+    :param path: Path to the log file.
+    """
+
+    def __init__(self, path):
+        super().__init__(os.path.basename(path))
+        self.path = path
+
+    def __repr__(self):
+        r = "sysinfo.Logfile(%r, %r)"
```

Comment:
s/sysinfo\.//g

Reply:
I imagine it will show up in the logs, so, `sysinfo.` here is a good hint of where the information came from.

Reply:
__repr__ is used to be a  representation of an object and  it is not usual to add prefix to that class. Otherwise  we should add this to all other __repr__  that we have. If we need that for log, my suggestion is to add the prefix when logging, not here.

Reply:
Okay, I lost it was inside `__repr__`.

Reply:
I was using the old code of sysinfo, but definitely we can do the updates here.